src/service-object.ts

```diff
@@ -49,12 +50,12 @@ export interface ServiceObjectConfig {
   /**
    * The base URL to make API requests to.
    */
-  baseUrl: string;
+  baseUrl?: string;
 
   /**
    * The method which creates this object.
    */
-  createMethod: Function;
+  createMethod?: Function;
```

Comment:
Is `createMethod` really optional? What if `create` is called if `createMethod` isn't defined?

Reply:
The extending class, e.g. `Bucket` from the Storage API, defines what methods it needs in the config: https://github.com/googleapis/nodejs-storage/blob/1837107d425cd2da6051598a6779418baa6a7a6f/src/bucket.js#L65

If it doesn't provide `create`, then it's not on the object: https://github.com/JustinBeckwith/nodejs-common/blob/425bbfd1ae76dd54e91a80bff362a02deab41b68/src/service-object.ts#L168.

Reply:
In my case, I just noticed (well, got a compiler error) because the tests didn't pass one in.  That made me assume it's actually optional :)

Reply:
Ah, that seems pretty tricky to pin down with types.